### DISCUSSION TITLE
Transfer Aptos Coin as a Fungible Asset

### DISCUSSION THREAD

**Discussion by 0xblockbard** [aptos-labs/aptos-developer-discussions#432] [Questions] [answered]
Hello, most of the docs and code examples show Aptos Coin with the Coin Standard (https://aptos.dev/en/build/smart-contracts/aptos-coin).

However, I've read that it has migrated over to the Fungible Asset Standard, and it's not very clear to me how we will interact with the Aptos Coin using the Fungible Asset Standard.

E.g. legacy coin standard:
use aptos_framework::coin;
coin::transfer(&signer, recipient, amount);

on Fungible Asset standard:
from docs: public entry fun primary_fungible_store::transfer<T: key>(sender: &signer, metadata: Object, recipient: address, amount:u64)

can we still use the legacy coin standard in our modules for interacting with Aptos Coin?
do we still use use aptos_framework::coin ?
how do we get the metadata of Aptos Coin if we use the fungible asset standard?
do we still need to support both FA types and Coin types in our modules? or just Fungible Asset types?

----
question moved from https://github.com/aptos-labs/aptos-developer-discussions/discussions/427

**Comment by kshitijc1506** [comment, marked as answer]
FA migration is live on mainnet so you can burn Coin APT and mint FA APT (in coin.move). Although, probably best to do this after there is CEX support for FA APT first. This is irreversible. 

1) Yes you can use the legacy coin standard
2) Yes
3) 0xA is the metadata object address
4) Ideally, you have both, keeping in mind that things will move completely to FA in the coming months / years. 

Heres a discussion for more context on implementation https://github.com/aptos-labs/aptos-developer-discussions/discussions/407

**Reply by MoonShiesty** [reply]
whats the ETA on CEX support for FA APT?

**Reply by kshitijc1506** [reply]
It's not fixed yet AFAIK, my best estimate is 3-6 months.
